torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -419,4 +447,26 @@ struct VarianceMeanOpRecord : RecordFunctor {
   bool keepdim_;
 };
 
+struct CopyToOpRecord : RecordFunctor {
+  CopyToOpRecord(std::vector<size_t> _args, std::vector<size_t> _outputs)
+      : RecordFunctor(std::move(_args), std::move(_outputs)) {}
+  virtual ~CopyToOpRecord() = default;
+
+  void operator()(FusionDefinition& fd) final {
+    auto dest = fd.getFusionState(args.at(0))->as<NvfTensorView>();
+    auto source = fd.getFusionState(args.at(1))->as<NvfTensorView>();
+
+    if (dest->isFusionInput()) {
+      fd.fusionPtr()->aliasOutputToInput(source, dest);
```

Comment:
Yes, I also thought about this and tried that. Something was wrong, I don't remember what exactly, I should try again.

This approach is also better because we wouldn't need to try conditionally removing tensors from fusion output and adding them back later in the correct order since `source` is now completely disconnected.

Reply:
`auto tmp = torch::jit::fuser::cuda::set(source);` is now used. My initial mistake was to use `tmp = set(source)` only for `dest->isFusionInput()` path, it becomes difficult to drop this output for the "else" branch.